snuba/web/split.py

```diff
@@ -328,6 +329,17 @@ def execute(
             metrics.increment("column_splitter.query_above_limit")
             return None
 
+        # Do not split if there is already a = or IN condition on an ID column
+        for expr in query.get_condition_from_ast() or []:
+            match = FunctionCall(
```

Comment:
nit: you can create the pattern in the constructor. It is an immutable object you do not need to recreate it at every call.